### PR TITLE
add \text to num_head in Mathematical formula

### PR DESCRIPTION
**Pull request by Zrealshadow**
Fixes #{49296}

In doc string of `nn.functional.py` Method:`multi_head_attention_forward`

Have to use `\text` to enclose the `num_heads`.

Otherwise it can not display normally

### CODE REVIEW

**Comment by mariosasko** [comment]
To properly link the issue, replace `Fixes #{49296}` with `Fixes #49296`

It should be `\text{num\_heads}` (notice the underscore) instead of `\text{num_heads}` to fully fix the formatting. 

@mruberry Can you please take it from here (to help him with signing a CLA)?

**Comment by Zrealshadow** [comment]
I use` \text{} `to  enclose `num_head`.
No need to add add escapes before underscore

**Review by mruberry — CHANGES_REQUESTED** [review]
Hi @Zrealshadow, thank you for trying to improve the documentation.

There seem to be numerous issues with the Multihead Attention documentation. For example, the sections being edited don't seem to appear in the current docs:

https://pytorch.org/docs/master/generated/torch.nn.MultiheadAttention.html?highlight=multiheadattention#torch.nn.MultiheadAttention

Or in the docs generated by this PR:

https://9584852-65600975-gh.circle-artifacts.com/0/docs/generated/torch.nn.MultiheadAttention.html?highlight=multi%20head%20attention%20forward#torch.nn.MultiheadAttention.forward

Where the shape should appear, for example, I just see "3D mask SHAPE_SHOULD_BE_HERE! where N is the batch size, L is the target sequence length" on my browser, minus the "SHAPE_SHOULD_BE_HERE!". 

Do you see the same thing on your browser?

**Comment by Zrealshadow** [comment]
Yes, in my browser, for above two docs, "3D mask {SHAPE_SHOULD_BE_HERE!} where N is the batch size, L is the target sequence length"
{SHAPE_SHOULD_BE_HERE!}  disappears.

**Comment by mruberry** [comment]
> Yes, in my browser, for above two docs, "3D mask {SHAPE_SHOULD_BE_HERE!} where N is the batch size, L is the target sequence length"
> {SHAPE_SHOULD_BE_HERE!} disappears.

So what's up with that? The mathematical expression this PR is attempting to correct doesn't even render!

**Comment by Zrealshadow** [comment]
> > Yes, in my browser, for above two docs, "3D mask {SHAPE_SHOULD_BE_HERE!} where N is the batch size, L is the target sequence length"
> > {SHAPE_SHOULD_BE_HERE!} disappears.
> 
> So what's up with that? The mathematical expression this PR is attempting to correct doesn't even render!

It is wired, this mathematical expression is valid in mathjax

**Comment by mruberry** [comment]
> > > Yes, in my browser, for above two docs, "3D mask {SHAPE_SHOULD_BE_HERE!} where N is the batch size, L is the target sequence length"
> > > {SHAPE_SHOULD_BE_HERE!} disappears.
> > 
> > 
> > So what's up with that? The mathematical expression this PR is attempting to correct doesn't even render!
> 
> this mathematical expression is valid in mathjax， It's kind of weird

I agree it looks valid to me, too. Does it render if you build the docs locally? Before proceeding on this PR, I think that's the issue we need to resolve. I'm curious what you discover.

**Comment by mariosasko** [comment]
@mruberry Sorry, forgot to mention that this PR changed formatting in `nn.functional.multi_head_attention_forward`, but this function doesn't have an entry in the docs currently. However, I tought it's worth fixing it in advance.

@Zrealshadow About the underscore, e.g. check the weight parameter of the quanized conv1d in the current docs: https://pytorch.org/docs/master/torch.nn.quantized.html#torch.nn.quantized.functional.conv1d
and then check the actual source:
https://pytorch.org/docs/master/_modules/torch/nn/quantized/functional.html#conv1d
https://github.com/pytorch/pytorch/blob/1a92802bde28286f850b415bfd985515565740c3/torch/nn/quantized/functional.py#L124

I gave you a single example, but there are more.
There was a PR without the underscore that didn't render properly, but I can't find it now.
It's hard to debug this, since there is no entry in the docs, so fixing nn.MultiHeadAttention's docs might be easier.

**Comment by Zrealshadow** [comment]
> @mruberry Sorry, forgot to mention that this PR changed formatting in `nn.functional.multi_head_attention_forward`, but this function doesn't have an entry in the docs currently. However, I tought it's worth fixing it in advance.
> 
> @Zrealshadow About the underscore, e.g. check the weight parameter of the quanized conv1d in the current docs: https://pytorch.org/docs/master/torch.nn.quantized.html#torch.nn.quantized.functional.conv1d
> and then check the actual source:
> https://pytorch.org/docs/master/_modules/torch/nn/quantized/functional.html#conv1d
> https://github.com/pytorch/pytorch/blob/1a92802bde28286f850b415bfd985515565740c3/torch/nn/quantized/functional.py#L124
> 
> I gave you a single example, but there are more.
> There was a PR without the underscore that didn't render properly, but I can't find it now.
> It's hard to debug this, since there is no entry in the docs, so fixing nn.MultiHeadAttention's docs might be easier.

According to your Example, I change this PR. Thank you

**Comment by mruberry** [comment]
Would you rebase this, @Zrealshadow? Looks like the doc build didn't get a chance to run.
